### PR TITLE
babel-plugin-macros #1200

### PR DESCRIPTION
**Pull request by PiereDome**
Prototype for create babel Macro issue #1200 

I added a babel-macro in the /src/macro folder and used the files from the babel-plugin-styled-components library that I needed. Be aware that there is a /* eslint-disable */ on all of the files. I wanted some feedback on this to see if there is anything glaring that I'm missing as I work to get the flow-types in and the linting fixes for the code brought in.

### CODE REVIEW

**Comment by PiereDome** [comment]
Ah, yes. That is a great idea. Will get that implemented and should save me from fixing the linting and specifying the flow types

**Comment by PiereDome** [comment]
Just added the changes and have linting and flow passing

**Comment by PiereDome** [comment]
Yeah, I'll work on that this evening

**Comment by PiereDome** [comment]
Built out the tests and started to build out fixtures for each option for the plugin and realized I might of started to get carried away. So only built out ssr and minify as different from default. Let me know if you want more or less tests. Oh and let me know if you'd like me to rebase as well.

**Comment by PiereDome** [comment]
@mxstbr I have built out a simple test that the plugin is doing what it should. Do you want to look and see if there is more you want from the tests

**Comment by philpl** [comment]
@mxstbr I haven’t picked up a review right from the start. Mind having another look?

**Comment by mxstbr** [comment]
This looks good code-wise, just waiting for news from @gaearon in https://github.com/facebookincubator/create-react-app/issues/2730

**Comment by philpl** [comment]
@mxstbr I have a project that’s set up with Babel-macros so I can give it a go later 😬

While we’re here, would it be worth moving the Babel plugin into the same repo eventually? That’d be helping me out a lot for v3 actually, and could ease migration in the future.

**Comment by mxstbr** [comment]
Yeah didn't we plan on using lerna? #588 or something

**Comment by philpl** [comment]
@mxstbr we did originally, but in general I think we could just add “styled-components/Babel” and keep the plugin src in this repo too. I’m not too sure about this but it might be helpful soon

**Comment by mxstbr** [comment]
Hmm, I guess so. Not sure I like that we keep adding `/primitives`, `/native`, `/no-parser` etc. :confused:

File: package.json [hunk 1]
```diff
@@ -78,9 +79,11 @@
     "babel-core": "^6.17.0",
     "babel-eslint": "^7.1.1",
     "babel-loader": "^6.2.10",
+    "babel-macros": "^1.1.1",
```
Comment (kentcdodds) [inline package.json]: Note: `babel-macros` has had a name change to improve compatibility with babel v7. It's now `babel-plugin-macros`. That's the only breaking change. It's currently on `2.0.0`.

**Comment by probablyup** [comment]
Is this still a thing?

**Comment by probablyup** [comment]
Going to close this for now since there hasn't been any movement in a long time.

**Comment by mxstbr** [comment]
Definitely still a thing, `create-react-app` will have built-in support for Babel macros as of `v2` so we should definitely ship this.

**Comment by TrySound** [comment]
How's it going? Can I help? I definitely need this feature right now.

**Comment by mxstbr** [comment]
Anybody want to pick this up? create-react-app v2 is happening soon, so we should add a macro so folks using cra can also use the styled-components Babel plugin!

**Comment by mareksuscak** [comment]
Can you share more details on what remains to be done here? I might take the plunge if there's no one else.

**Comment by probablyup** [comment]
#2032 picks this up and finishes it, so closing

**Pull request closed without merging**